mesonbuild/modules/fs.py

```diff
@@ -310,6 +313,18 @@ def copyfile(self, state: ModuleState, args: T.Tuple[FileOrString, T.Optional[st
 
         return ModuleReturnValue(ct, [ct])
 
+    @FeatureNew('fs.relative_to', '1.3.0')
+    @typed_pos_args('fs.relative_to', (str, File), (str, File))
+    def relative_to(self, state: ModuleState, args: T.Tuple[FileOrString, FileOrString], kwargs: 'ReadKwArgs') -> str:
+        t = str(args[0])
+        f = str(args[1])
```

Comment:
`str()` of File object returns its relative path, not sure that's what is desired here. Maybe we should error out if they are not both absolute paths? Otherwise we get weird behaviours:
```
>>> os.path.relpath('foo', '/bar')
'../home/xclaesse/foo'
```

Reply:
Or assume that relative paths in arguments are relative to current_source_dir(). I think Python assumes they are relative to current workdir which makes little sense for Meson.

Reply:
I like that idea. I will let others comment on it before pushing a v2

Reply:
IMHO it makes sense to write `fs.relative_to('foo', meson.global_build_root())` when `foo` is a file in current source directory that you pass to a custom_target() that runs in builddir.